File: 35. Search Insert Position/Search Insert Position.cpp
#include<iostream>
#include<vector>
#include<unordered_map>

using namespace std;

int searchInsert(vector<int>& nums, int target) 
{
    int Size = nums.size();
    if(Size == 0) return 0;
    for (int i = 0; i < nums.size(); i++)
    {
        if(nums[i] >= target )return i;
    }
    return Size;  
}

int main()
{

}

File: 35. Search Insert Position/Search Insert Position.py
from typing import List


def search(nums: List[int], target: int) -> int:
    L = 0
    R = len(nums) - 1
    while R >= L:
        Mid = L + ((R - L) // 2)
        if nums[Mid] == target:
            return Mid
        elif nums[Mid] > target:
            R = Mid - 1
        else:
            L = Mid + 1
    return R + 1


print(search([1, 3, 5, 6], -1))
